File: core/src/bara/game/components/PhysicsComponent.java
package bara.game.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.physics.box2d.Body;
import com.badlogic.gdx.physics.box2d.BodyDef;
import com.badlogic.gdx.physics.box2d.Fixture;
import com.badlogic.gdx.physics.box2d.FixtureDef;
import com.badlogic.gdx.utils.Pool;
import bara.game.systems.WorldRenderSystem;

import java.util.HashMap;

public class PhysicsComponent implements Component, Pool.Poolable {
    public BodyDef bodyDef = null;
    public HashMap<String, FixtureDef> fixtureDefs = new HashMap<String, FixtureDef>();
    public Body body = null;
    public HashMap<String, Fixture> fixtures = new HashMap<String, Fixture>();

    @Override
    public void reset() {
        bodyDef = null;
        fixtureDefs.clear();
        body = null;
        fixtures.clear();
    }

}

File: core/src/bara/game/components/PhysicsComponent.kt
package bara.game.components

import bara.game.ashley.GameComponent
import com.badlogic.gdx.physics.box2d.Body
import com.badlogic.gdx.physics.box2d.BodyDef
import com.badlogic.gdx.physics.box2d.Fixture
import com.badlogic.gdx.physics.box2d.FixtureDef
import com.badlogic.gdx.utils.ObjectMap
import bara.game.ashley.EngineEntity

class PhysicsComponent : GameComponent() {
    var bodyDef: BodyDef? = null
    var fixtureDefs = ObjectMap<String, FixtureDef>()
    var body: Body? = null
    var fixtures = ObjectMap<String, Fixture>()

    override fun reset() {
        bodyDef = null
        fixtureDefs.clear()
        body = null
        fixtures.clear()
    }

    override fun buildOnto(engineEntity: EngineEntity) {
        val tempComponent = this
        engineEntity.with<PhysicsComponent> {
            bodyDef = tempComponent.bodyDef
            tempComponent.fixtureDefs.forEach { entry ->
                fixtureDefs.put(entry.key, entry.value)
            }
        }
    }

}
